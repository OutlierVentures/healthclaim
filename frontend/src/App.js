import React, { Component, Fragment } from "react";
import { LinkContainer } from "react-router-bootstrap";
import { Link, withRouter } from "react-router-dom";
import { Nav, Navbar, NavItem } from "react-bootstrap";
import Routes from "./Routes";
import "./App.css";

const { Issuer } = require('openid-client');
const crypto = require('crypto');
var client_here = undefined;

class App extends Component {
  constructor(props) {

    super(props);

    this.state = {
      isAuthenticated: props.isAuthenticated,
      isAuthenticating: true,
      result: undefined,
      client: undefined,
      callback_url: undefined
    };

  }

  async componentDidMount() {
    this.setState({ isAuthenticating: false });
  }


  userHasAuthenticated = authenticated => {
    this.setState({ isAuthenticated: authenticated });
  }

  userHasResult = result => {
    result = JSON.parse(result);
    this.setState({ result: result });
  }

  updateClient = client => {
    this.setState({ client });
  }

  updateCallbackUrl = callback_url => {
    this.setState({ callback_url });
  }

  updateAccess_token = access_token => {
    this.setState({ access_token });
  }

  updateEmail = email => {
    this.setState({ email });
  }

  handleLogout = async event => {
    this.userHasAuthenticated(false);
    this.props.history.push("/login");
  }

  render() {
    const childProps = {
      isAuthenticated: this.state.isAuthenticated,
      userHasAuthenticated: this.userHasAuthenticated,
      result: this.state.result,
      userHasResult: this.userHasResult,
      updateClient: this.updateClient,
      client: this.state.client,
      callback_url: this.state.callback_url,
      updateCallbackUrl: this.updateCallbackUrl,
      access_token: this.state.access_token,
      updateAccess_token: this.updateAccess_token,
      email: this.state.email,
      updateEmail: this.updateEmail
    };

    return (
      !this.state.isAuthenticating &&
      <div className="App container">
        <Navbar fluid collapseOnSelect>
          <Navbar.Header>
            <Navbar.Brand>
              <Link to="/">OPAL Credit</Link>
            </Navbar.Brand>
            <Navbar.Toggle />
          </Navbar.Header>
          <Navbar.Collapse>
            <Nav pullRight>
              {this.state.isAuthenticated
                ? <Fragment>
                  <LinkContainer to="/score">
                    <NavItem>Credit Score</NavItem>
                  </LinkContainer>
                  <NavItem onClick={this.handleLogout}>Logout</NavItem>
                </Fragment>
                : <Fragment>
                  <LinkContainer to="/signup">
                    <NavItem>Signup</NavItem>
                  </LinkContainer>
                  <LinkContainer to="/login">
                    <NavItem>Login</NavItem>
                  </LinkContainer>
                </Fragment>
              }
            </Nav>
          </Navbar.Collapse>
        </Navbar>
        <Routes childProps={childProps} />
      </div>
    );
  }
}

export default withRouter(App);