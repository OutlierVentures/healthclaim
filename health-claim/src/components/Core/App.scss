// App.scss

// Scaffo
html {
  background: #f2f2f2;
}
.Body {
  background: #f2f2f2;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.Wrapper {
  height: 100vh;
  display: flex;
  flex-direction: column;
}

.Main {
  // max-width: 960px;
  margin: 0 auto;
  padding-bottom: 96px;
  z-index: 2;
}

.Page__Title {
  padding-top: 1.67em;
  margin-bottom: 0.33em;
  // position: relative;
  // display: block;
  .Page__Title--deadline {
    float: right;
    position: absolute;
    font-size: 0.67em;
    padding-top: 0.51em;
    line-height: 1;
    right: 0;
    opacity: 0.7;
  }
}
.Page__Subtitle {
  margin: 0;
}
.Section__Header {
  font-size: 16px;
  text-transform: uppercase;
  font-weight: bold;
  color: rgba(48, 94, 212, 1);
  padding-top: 1.7em;
  padding-bottom: 0.5em;
  margin-bottom: 0.67em;
  border-bottom: 1px solid #e0e0e0;
}

// Temp
.Header {
  display: flex;
  flex-direction: column;
}
.Header__Layout {
  display: flex;
  width: 100%;
  flex-direction: row;
  justify-content: space-between;
}

.Header__Upper {
  background: rgb(28, 29, 92);
  height: 56px;
  z-index: 99;
  width: 100%;
  .Header__Nav {
    align-items: center;
    display: flex;
    height: 56px;
    margin-left: 48px;
    margin-right: 48px;
    // max-width: 960px;
  }
  .Header__Link,
  .Header__Logo {
    color: #fff;
    font-weight: 600;
    margin: 0 1vw;
    & > a {
      color: #fff;
      text-decoration: none;
    }
  }
  .Header__Logo {
    opacity: 0.66;
  }
}

.Header__Link {
  margin-left: 35px;
}

.Header__Link__text {
  color: white;

  &:hover {
    color: lightblue;
  }
}

.Header__Lower {
  align-items: center;
  background: rgba(48, 94, 212, 1);
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-content: center;
  min-height: 56px;
  padding-left: 48px;
  padding-right: 48px;

  height: 56px;
  z-index: 3;
  width: 100%;
  .Header__Title {
    color: #fff;
    line-height: 1;
    margin-top: -4px;
    margin-bottom: 0;
    margin-left: 1vw;
    font-size: 1em;
  }
  a {
    color: white;
  }
}

.video_background {
  /* class name used in javascript too */
  min-width: 100%;
  min-height: 100%;
  width: auto;
  height: auto;
  position: absolute;
  left: 0;
  top: 0;
  z-index: 0;
}
