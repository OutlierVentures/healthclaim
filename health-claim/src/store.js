import { combineReducers } from "redux";
import { createStore, applyMiddleware, compose } from "redux";
import thunkMiddleware from "redux-thunk";
import { createLogger } from "redux-logger";
import storage from "redux-persist/lib/storage";
import { persistStore, persistReducer } from "redux-persist";

import fetchNotifications from "./reducers/fetchNotifications";
import fetchPatientClaims from "./reducers/fetchPatientClaims";
import selectPatient from "./reducers/selectPatient";
import login from "./reducers/login";
import fetchCredDef from "./reducers/fetchCredDef";
import register from "./reducers/register";
import verifySession from "./reducers/verifySession";
import sessionToken from "./reducers/sessionToken";
import logout from "./reducers/logout";
import setPredicates from "./reducers/setPredicates";
import createMasterSecret from "./reducers/createMasterSecret";
import storeRecord from "./reducers/storeRecord";
import fetchCreds from "./reducers/fetchCredentials";
import thirdPartyCalls from "./reducers/thirdPartyCalls";
import fetchCachedCreds from "./reducers/fetchCachedCreds";
import fetchCredReqStatus from "./reducers/fetchCredReqStatus";

const loggerMiddleware = createLogger();

const persistConfig = {
  key: "root",
  storage,
  whitelist: ["sessionToken", "setPredicates"]
};

const reducer = combineReducers({
  fetchPatientClaims,
  fetchNotifications,
  selectPatient,
  login,
  setPredicates,
  logout,
  register,
  verifySession,
  sessionToken,
  createMasterSecret,
  storeRecord,
  fetchCredDef,
  fetchCreds,
  thirdPartyCalls,
  fetchCachedCreds,
  fetchCredReqStatus
});

const persistedReducer = persistReducer(persistConfig, reducer);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default () => {
  let store = createStore(
    persistedReducer,
    composeEnhancers(
      applyMiddleware(
        thunkMiddleware, // lets us dispatch() functions
        loggerMiddleware // neat middleware that logs actions
      )
    )
  );
  let persistor = persistStore(store);
  return {
    store,
    persistor
  };
};
