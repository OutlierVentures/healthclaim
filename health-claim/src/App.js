import React from 'react';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Login from "./containers/Login";
import Doctor from "./containers/Doctor";
import Claim from "./containers/Claim";
import NotFound from "./containers/NotFound";
import './App.css';

export class App extends React.PureComponent {
  render() {
    return (
      <div className="App container">
        <Router>
          <Switch>
            <Route path="/" exact component={Login} />
            <Route path="/claim" exact component={Claim} />
            <Route path="/doctor" exact component={Doctor} />
            <Route path="/login" exact component={Login} />
            { /* Finally, catch all unmatched routes */}
            <Route component={NotFound} />\
          </Switch>
        </Router>
      </div>
    );
  }
}

export default App;
