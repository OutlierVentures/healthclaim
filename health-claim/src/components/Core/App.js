import React from "react";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.css";
import "./App.scss";
import RoutesContainer from "../../RoutesContainer";

import SessionAuthenticator from "../../containers/SessionAuthenticatorContainer";

export class App extends React.PureComponent {
  render() {
    return (
      <SessionAuthenticator>
        <Router>
          <div>
            <link
              rel="stylesheet"
              href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
            />
            <div className="Header">
              <div className="Header__Upper">
                <div className="Header__Nav">
                  <div className="Header__Layout">
                    <div className="Header__Logo">HealthClaim</div>{" "}
                    {this.props.loggedIn ? (
                      <div className="Header__Link">
                        <Link to="/" onClick={() => this.props.logout()}>
                          Logout
                        </Link>
                      </div>
                    ) : null}
                  </div>
                </div>
              </div>
              <div className="Header__Lower">
                <h2 className="Header__Title">
                  <Link style={{ textDecoration: "none" }} to="/dashboard">
                    Decentralized Healthcare
                  </Link>
                </h2>
                {this.props.loggedIn
                  ? [
                      <div className="Header__Link" key="1">
                        <Link className="Header__Link__text" to="/dashboard">
                          Dashboard
                        </Link>
                      </div>,
                      <div className="Header__Link" key="2">
                        <Link className="Header__Link__text" to="/settings">
                          Settings
                        </Link>
                      </div>
                    ]
                  : null}
              </div>
            </div>
            <div className="Body">
              <div className="Main">
                <Route component={RoutesContainer} />
              </div>
            </div>
          </div>
        </Router>
      </SessionAuthenticator>
    );
  }
}

export default App;
