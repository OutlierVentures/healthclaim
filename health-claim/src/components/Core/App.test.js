import React from "react";
// import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow"; // ES6
import App from "./App";
import store from "../../store";
describe("App", () => {
  const renderer = new ShallowRenderer();
  it("should render without crashing", () => {
    const tree = renderer.render(<App store={store} />);
    expect(tree).toMatchSnapshot();
  });
});
