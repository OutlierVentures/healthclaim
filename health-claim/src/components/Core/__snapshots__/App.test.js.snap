// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<withRouter(Connect(SessionAuthenticator))>
  <BrowserRouter>
    <div>
      <link
        href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
        rel="stylesheet"
      />
      <div
        className="Header"
      >
        <div
          className="Header__Upper"
        >
          <div
            className="Header__Nav"
          >
            <div
              className="Header__Layout"
            >
              <div
                className="Header__Logo"
              >
                HealthClaim
              </div>
               
            </div>
          </div>
        </div>
        <div
          className="Header__Lower"
        >
          <h2
            className="Header__Title"
          >
            Decentralized Healthcare
          </h2>
        </div>
      </div>
      <div
        className="Body"
      >
        <div
          className="Main"
        >
          <Route
            component={[Function]}
          />
        </div>
      </div>
    </div>
  </BrowserRouter>
</withRouter(Connect(SessionAuthenticator))>
`;
